core/services/ocr2/plugins/functions/plugin.go

```diff
@@ -81,6 +86,38 @@ func NewFunctionsServices(sharedOracleArgs *libocr2.OCR2OracleArgs, conf *Functi
 	}
 	allServices = append(allServices, job.NewServiceAdapter(functionsReportingPluginOracle))
 
+	s4OracleArgs := *sharedOracleArgs
+	s4OracleArgs.ReportingPluginFactory = s4.S4ReportingPluginFactory{
+		Logger: sharedOracleArgs.Logger,
+		ORM:    s4ORM,
+		ConfigDecoder: func(configBytes []byte) (*s4.PluginConfig, *types.ReportingPluginLimits, error) {
+			wrapper, err2 := config.DecodeReportingPluginConfig(configBytes)
+			if err != nil {
+				return nil, nil, err2
+			}
+			if wrapper.Config == nil || wrapper.Config.S4PluginConfig == nil {
+				return nil, nil, errors.New("missing s4 plugin config")
+			}
+			return &s4.PluginConfig{
```

Comment:
Nice but there is a bit of a discrepancy with how TDH2 plugin handles config - it packs it into a single structure and later unpacks in NewReportingPlugin (https://github.com/smartcontractkit/tdh2/blob/main/go/ocr2/decryptionplugin/decryption.go#L36).
We should try to unify those two approaches. The one here seems to be cleaner.